### PR TITLE
Debot invoke impl

### PR DESCRIPTION
**Pull request by Keshoid**
Support for debot invoking in DEngine. 
- filter invoke calls from debot messages.
- send invoke msgs to browser.
- redesign test browser to store several debots with single message queue.
Modify `encode_internal_message` - support building of messages with empty body.

### CODE REVIEW

File: ton_client/src/debot/mod.rs [hunk 1]
```diff
@@ -216,12 +214,7 @@ pub fn remove(context: Arc<ClientContext>, params: RegisteredDebot) -> ClientRes
 pub struct ParamsOfSend {
     /// Debot handle which references an instance of debot engine.
     pub debot_handle: DebotHandle,
-    /// Std address of interface or debot.
-    pub source: String,
-    /// Function Id to call
-    pub func_id: u32,
-    /// Json string with parameters
-    pub params: String,
+    pub message: String,
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/mod.rs:217]: Add comment to `message` field

File: ton_client/src/debot/run_output.rs [hunk 2]
```diff
@@ -70,14 +77,38 @@ impl RunOutput {
...
+        Some(msg)
+    }
+
+    fn filter_invoke_call<'a>(
+        &mut self,
+        msg: (&'a Message, String),
+    ) -> Option<(&'a Message, String)> {
+        if msg.0.is_internal() {
+            let wc_id = msg.0.workchain_id().unwrap_or(BASECHAIN_ID);
+            if wc_id == BASECHAIN_ID || wc_id == MASTERCHAIN_ID {
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/run_output.rs:105]: I suggest check for workchain IDs resrved to debots instead of enumerating all existing workchains

**Review by AlexeyVavilin — CHANGES_REQUESTED** [review]

File: ton_client/src/debot/mod.rs (hunk 1 above)
Comment (Keshoid) [reply ton_client/src/debot/mod.rs:217]: Done

File: ton_client/src/debot/run_output.rs (hunk 2 above)
Comment (Keshoid) [reply ton_client/src/debot/run_output.rs]: Done

File: ton_client/src/debot/mod.rs (hunk 1 above)
Comment (AlexeyVavilin) [inline ton_client/src/debot/mod.rs:218]: BOC should be all capital

Comment (AlexeyVavilin) [inline ton_client/src/debot/mod.rs:218]: BOC should be all capital

**Review by AlexeyVavilin — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,15 +5,23 @@ All notable changes to this project will be documented in this file.
 ## [1.10.0] – 2021-02-28
 
 ### New
-
-- `boc.get_blockchain_config` function can extract blockchain configuration from key block and also 
+- Add optional field `src_address` to `ParamsOfEncodeInternalMessage`.
```
Comment (AlexeyVavilin) [inline CHANGELOG.md:8]: `abi` field became optional

File: ton_client/src/debot/mod.rs (hunk 1 above)
Comment (Keshoid) [reply ton_client/src/debot/mod.rs:218]: Ok

File: CHANGELOG.md (hunk 3 above)
Comment (Keshoid) [reply CHANGELOG.md:8]: Ok

**Pull request merged**